### PR TITLE
Fix GitHub token config update

### PR DESCRIPTION
**Pull request by yhaliaw**
When the github token config is updated, the token used by policy-service-compliance is not updated. This PR fixes this issue.

### CODE REVIEW

**Review by jdkandersson — DISMISSED** [review]

File: src/charm.py [hunk 1]
```diff
@@ -674,7 +674,8 @@ def _start_services(self) -> None:
         )
         self.repo_check_systemd_service.write_text(service_content, encoding="utf-8")
 
-        execute_command(["/usr/bin/systemctl", "start", "repo-policy-compliance"])
+        execute_command(["/usr/bin/systemctl", "daemon-reload"])
+        execute_command(["/usr/bin/systemctl", "restart", "repo-policy-compliance"])
```
Comment (jdkandersson) [inline src/charm.py:678]: I think this means that any existing tokens for spawned runners will be invalid, so we need to make sure to remove existing runners as well (perhaps this is already being done as a part of changing the token)

**Review by jdkandersson — DISMISSED** [review]

**Review by jdkandersson — DISMISSED** [review]

**Review by jdkandersson — APPROVED** [review]

**Review by arturo-seijas — APPROVED** [review]
LGTM

**Review by mthaddon — COMMENTED** [review]
We should really avoid using stored state for this (or anything else if possible). Is there some other way we can determine if the token has changed by inspecting the live state of the service, for instance?

**Comment by yhaliaw** [comment]
> We should really avoid using stored state for this (or anything else if possible). Is there some other way we can determine if the token has changed by inspecting the live state of the service, for instance?

I think this is planned as a part of the charm review and refactor.

**Pull request merged**